pkg/client/scout/reporter_test.go

```diff
@@ -256,9 +257,17 @@ func TestInstallID(t *testing.T) {
 		}
 	}
 	origEnv := os.Environ()
+	version.Version = "testing"
+	defer func() { version.Version = "" }()
+
 	for tcName, tcData := range testcases {
 		tcData := tcData
 		t.Run(tcName, func(t *testing.T) {
+			if tcData.InputGOOS != runtime.GOOS {
```

Comment:
Is this what makes the test fail when using Go 1.18? A short description of the fix in the commit description would be nice ;-)

Reply:
Just tested with Go 1.18.1. That fails miserably without this PR.

Reply:
I installed Go 1.18.1 on my mac. It fails there too, so this isn't linux specific.

Reply:
It's strange that it would pass for you and in CI since that would mean that non-linux tests are being run under Linux which one would expect to fail...

Reply:
I'm fairly certain that this is related to Go 1.18 (we use 1.17 in CI). It runs fine with 1.17 on all platforms. It breaks with 1.18 on all platforms.